### DISCUSSION TITLE
Maintain call stack group expand/collapse states

### DISCUSSION THREAD

**Discussion by jasonLaster** [RecordReplay/devtools#6398] [Ideas]
The challenge with maintaining expand/collapse states is that we'd want to do this with some knowledge of the stack. 

One short term option is to allow disabling framework grouping

**Comment by jasonLaster** [comment]
This is similar to also maintaining expand/collapse states for scopes and objects which would also be hard without persistent object ids

**Reply by markerikson** [reply]
Is each step a different "pause point"?  If not, we could tie the expanded value into the lifetime of the current pause point and reset when that changes.  If each step is a different "pause", though, that won't work.

**Comment by jaril** [comment]
I'm down for disabling framework grouping altogether — saves us from a whole host of bugs and at worst keeps us at parity with chrome devtools.

**Reply by markerikson** [reply]
I'd certainly like to keep it if possible. I've always used Chrome for debugging, but that was one of the really neat things I liked about FF's DevTools.

**Reply by jasonLaster** [reply]
It's definitely a feature a lot of people enjoy which makes me reluctant to remove it outright

**Reply by jaril** [reply]
RIGHT — I hesitated writing that comment because I remembered @markerikson being wowed by it on the very first replay stream he did. 

The only reason I remember is because you attributed it to work that "the replay team did" and in my head i was like…

<img src="https://media.giphy.com/media/Ow59c0pwTPruU/giphy.gif"/>

**Reply by markerikson** [reply]
I did know that was in FF already :)

**Reply by jasonLaster** [reply]
oh snap - this already works.

<img width="724" alt="Screen Shot 2022-04-15 at 8 20 03 AM" src="https://user-images.githubusercontent.com/254562/163588751-af3656c1-a7c3-4db7-8e74-f277a7f27960.png">
